src/libPersistence/ContractStorage2.cpp

```diff
@@ -1021,46 +1127,89 @@ void ContractStorage2::InitTempState(bool callFromExternal) {
   }
 }
 
-dev::h256 ContractStorage2::GetContractStateHashCore(const dev::h160& address,
-                                                     bool temp) {
-  if (IsNullAddress(address)) {
-    LOG_GENERAL(WARNING, "Null address rejected");
-    return dev::h256();
-  }
+dev::h256 ContractStorage2::UpdateContractTrieCore(
```

Comment:
Just to confirm: we have removed direct data hashing and will use trie hashing regardless of state entries count?

Reply:
No we will still direct hash the states for contracts with map, which can be identified by fetching the metadata parsed for remote state fetching. Thanks for reminding, this part of logic however has not been implemented yet, will do ASAP.